Source/Constraint.swift

```diff
@@ -362,7 +362,7 @@ public class Constraint {
         // special logic for updating
         if updateExisting {
             // get existing constraints for this view
-            let existingLayoutConstraints = reverse(layoutFrom!.snp_installedLayoutConstraints)
+            let existingLayoutConstraints = Set<LayoutConstraint>(layoutFrom!.snp_installedLayoutConstraints)
```

Comment:
Regarding to your question: Yes Swift uses the proper equality check. Actually `Hashable` implements `Equatable` because of this. 

Having said that when the object that you insert into the set is an `NSObject` and not a swift object, `isEqual` gets called instead of the overload operator. I'm gonna revert this one.